pkg/compose/create.go

```diff
@@ -1095,8 +1093,6 @@ func buildMount(project types.Project, volume types.ServiceVolumeConfig) (mount.
 
 	bind, vol, tmpfs := buildMountOptions(volume)
 
-	volume.Target = path.Clean(volume.Target)
```

Comment:
this was the cause for https://github.com/docker/compose/issues/11345
volume being a func parameter, original struct isn't updated here and we eventually compare `/logs/` with `/logs`